WordPress/gutenberg
# Font Family and Font Face REST API endpoints: better data handling and errors

## What?

- Updates data handling for Font Family and Font Face REST API endpoints with better validation and error handling
- Adds several tests to cover validation and errors

## Why?

Part of https://github.com/WordPress/gutenberg/issues/55278

## Testing Instructions

Reviewing automated tests and the error cases that are handled is the easiest way to see all of the changes, including

- Handling malformed JSON when creating items, or when post_content is not valid json
- Extra settings that might be in the post_content are not returned
- `fontFamily` settings are sanitized for valid css variable creation
- Ensuring required settings are not missing, are empty, and have the correct type
- Handling quirks with `rest_validate_value_from_schema` and `oneOf` schema types
- Ensuring only font file types are valid for uploading font face files
